jedi/api/__init__.py

```diff
@@ -720,8 +720,12 @@ def __init__(self, code, namespaces, **kwds):
             if not isinstance(environment, InterpreterEnvironment):
                 raise TypeError("The environment needs to be an InterpreterEnvironment subclass.")
 
-        super().__init__(code, environment=environment,
-                         project=Project(Path.cwd()), **kwds)
+        if "project" in kwds:
+            super().__init__(code, environment=environment, **kwds)
+        else:
+            super().__init__(code, environment=environment,
+                             project=Project(Path.cwd()), **kwds)
+            
```

Comment:
Just use the signature `__init__(self, code, namespaces, *, project=None, **kwds)` 

and use an 

```
if project is None:
    project = Project(Path.cwd())
```

I feel like this is cleaner.